src/master-playlist-controller.js

```diff
@@ -889,17 +895,29 @@ export class MasterPlaylistController extends videojs.EventTarget {
       // it, instead of erring the player or retrying this playlist, clear out the current
       // blacklist. This allows other playlists to be attempted in case any have been
       // fixed.
-      videojs.log.warn('Removing all playlists from the blacklist because the last ' +
-                       'rendition is about to be blacklisted.');
+      let reincluded = false;
+
       playlists.forEach((playlist) => {
-        if (playlist.excludeUntil !== Infinity) {
+        // skip current playlist which is about to be blacklisted
+        if (playlist === currentPlaylist) {
+          return;
+        }
+        const excludeUntil = playlist.excludeUntil;
+
+        if (typeof excludeUntil !== 'undefined' && excludeUntil !== Infinity) {
```

Comment:
Adding your comment as a comment:

```suggestion
        // a playlist is not unblacklisted if it wasn't blacklisted to begin with.
        if (typeof excludeUntil !== 'undefined' && excludeUntil !== Infinity) {
```